### DISCUSSION TITLE
Reccomendation for a Board and Development Kit for a WSN Lab

### DISCUSSION THREAD

**Discussion by deshpandeyash** [contiki-ng/contiki-ng#2710] [General]
We are a research group from the Technical University of Munich and conduct a Lab called as the Wireless Sensor Networks Lab. 
For the last 10 years, the students developed different projects on Zoleria Remotes and Contiki. Its a very successful lab and had many tutorials for building a cool network. 

Alas, we realize that we are too late with moving on. We cannot find Zolertia Remotes anymore to buy online and replace the broken ones. We also realize that we are depriving the students of using a more latest OS with Contiki-NG. 

While some of our researchers use Contiki-NG for their experiments, they still use the old Zoleria Remotes. Therefore, there is no expertise in our group with trying newer platforms and hardware. 

Hence, I would like to ask the community what their opinion is on which hardware to move on to. We would change the tutorials from Conitki to Contiki-NG and change the hardware at the same time. 

We have looked online at the TI CC2538 DK. However, the board with the SmartRF06 and the large LCD screen seems bulky and also expensive. Can we do the lab with just the RF modules from this DK ? 

If not, which other platforms/hardware would be the best for us ?

**Comment by mbaddeley** [comment]
Most of the WSN community has moved on to Nordic devices these days.
Specifically, NG has support for the 52840 and 5340.

**Reply by deshpandeyash** [reply]
Thanks a lot for your input.

**Comment by rahav** [comment]
I do doctoral research at [Dr. Chenyang Lu lab](https://engineering.wustl.edu/faculty/Chenyang-Lu.html) and prior to that ten years as CTO, so I have a little understanding of good engineering.

I studied the landscape two months ago, for the same purpose you have. We had a very successful testbed, predominately built with TelosB. I concluded that there are two wordy options at this time. Nordic and Gecko. I randomly decided to explore the viability of Nordic first, and I have been happy with the nRF52840 family. Two devices are supported in this family. A large development board ([the nRF52840](https://www.nordicsemi.com/Products/Development-hardware/nrf52840-dk), officially called development kit or DK), and a small device called the [nRF52840 Dongle](https://www.nordicsemi.com/Products/Development-hardware/nrf52840-dongle).

This is [Contiki's tutorial](https://docs.contiki-ng.org/en/develop/doc/platforms/nrf52840.html) for these devices.

There are some pending issues, and I have been receiving EXCELLENT support from the Contiki-NG development team. As of today, one critical item is pending before I would be comfortable committing budget and time to build a new testbed. I am certain this last matter (#2704) will be resolved (it could be due to our setting and not with the Nordic hardware or Contiki).

Based on my experience, I cannot recommend the nRF5340 at this time. I don't think that the Contiki `TARGET` is there, to the level it is for the nRF52840.

We covered a lot of grounds and pain during the study, and will be happy to share more details, or collaborate with your group.

**Reply by deshpandeyash** [reply]
Hi. Thanks a lot for your input. We would like to use it for a higher bachelor, early master level course. Usually, the course is basic and covers the basics of WSN, and the students in the end have to build a working project of their choice that satisfies certain criteria. I am not sure if I can convince my colleagues to migrate their testbed but if the feedback from our experience is positive, I am sure they would love to move to newer hardware. 

Based on the answers, we have decided to purchase a few boards and begin our evaluation. 

I will definitely get back to you to get more inputs.

**Comment by deshpandeyash** [comment]
We bought the Nordic Boards and are fairly confident that we can move the Lab Course material from Zolertia remotes to these boards. 
Thanks for your inputs.
